packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -496,6 +496,15 @@ export abstract class SharedObjectCore<
 					localOpMetadata,
 				);
 			},
+			processMessages: (
+				messagesWithMetadata: {
+					message: ISequencedDocumentMessage;
+					localOpMetadata: unknown;
+				}[],
+				local: boolean,
+			) => {
+				this.processMessages(messagesWithMetadata, local);
+			},
```

Comment:
We should take this opportunity to move away from wrapping and re-wrapping the data in ISequencedDocumentMessage. It may have to be superficial to start, as it could be hard to do at higher levels, but the farther you can push it up the chain the better

```suggestion
			processMessages: (
				message: Omit<ISequencedDocumentMessage,"contnets">,
				messagesWithMetadata: {
					content: unknown
					localOpMetadata: unknown;
				}[],
				local: boolean,
			) => {
				this.processMessages(messagesWithMetadata, local);
			},
```

Reply:
Updated the logic to do this. Please take a look. The `clientSequenceNumber` is also different for these messages so I moved that to the content array as well.